Test/Unit/Cron/Providers/ProcessedOldNotificationsProviderTest.php

```diff
@@ -0,0 +1,147 @@
+<?php
+/**
+ *
+ * Adyen Payment module (https://www.adyen.com/)
+ *
+ * Copyright (c) 2025 Adyen N.V. (https://www.adyen.com/)
+ * See LICENSE.txt for license details.
+ *
+ * Author: Adyen <magento@adyen.com>
+ */
+
+namespace Adyen\Payment\Test\Cron\Providers;
+
+use Adyen\Payment\Api\Repository\AdyenNotificationRepositoryInterface;
+use Adyen\Payment\Cron\Providers\ProcessedOldNotificationsProvider;
+use Adyen\Payment\Helper\Config;
+use Adyen\Payment\Logger\AdyenLogger;
+use Adyen\Payment\Test\Unit\AbstractAdyenTestCase;
+use Magento\Framework\Api\SearchCriteria;
+use Magento\Framework\Api\SearchCriteriaBuilder;
+use Magento\Framework\Api\SearchResultsInterface;
+use Magento\Framework\Exception\LocalizedException;
+use Magento\Store\Api\Data\StoreInterface;
+use Magento\Store\Model\StoreManagerInterface;
+use PHPUnit\Framework\MockObject\MockObject;
+
+class ProcessedOldNotificationsProviderTest extends AbstractAdyenTestCase
+{
+    protected ?ProcessedOldNotificationsProvider $notificationsProvider;
+    protected AdyenNotificationRepositoryInterface|MockObject $adyenNotificationRepositoryMock;
+    protected SearchCriteriaBuilder|MockObject $searchCriteriaBuilderMock;
+    protected Config|MockObject $configHelperMock;
+    protected StoreManagerInterface|MockObject $storeManagerMock;
+    protected AdyenLogger|MockObject $adyenLoggerMock;
+
+    const STORE_ID = PHP_INT_MAX;
+
+    protected function setUp(): void
+    {
+        $this->adyenNotificationRepositoryMock =
+            $this->createMock(AdyenNotificationRepositoryInterface::class);
+        $this->searchCriteriaBuilderMock = $this->createMock(SearchCriteriaBuilder::class);
+        $this->configHelperMock = $this->createMock(Config::class);
+        $this->storeManagerMock = $this->createMock(StoreManagerInterface::class);
+        $this->adyenLoggerMock = $this->createMock(AdyenLogger::class);
+
+        $storeMock = $this->createMock(StoreInterface::class);
+        $storeMock->method('getId')->willReturn(self::STORE_ID);
+        $this->storeManagerMock->method('getStore')->willReturn($storeMock);
+
+        $this->notificationsProvider = new ProcessedOldNotificationsProvider(
+            $this->adyenNotificationRepositoryMock,
+            $this->searchCriteriaBuilderMock,
+            $this->configHelperMock,
+            $this->storeManagerMock,
+            $this->adyenLoggerMock
+        );
+    }
+
+    protected function tearDown(): void
+    {
+        $this->notificationsProvider = null;
+    }
+
+    public function testProvideSuccess()
```

Comment:
Is this method missing an assertion?

Reply:
Good catch, thank you! 

Return value assertions have been added in [this](https://github.com/Adyen/adyen-magento2/pull/2837/commits/5b7d8e82a7141cb1bfdaacd0b78837b5586043a7) commit.